src/Orleans/Messaging/SocketManager.cs

```diff
@@ -112,32 +121,24 @@ internal static void WriteConnectionPreemble(Socket socket, GrainId grainId)
         // We start an asynch receive, with this callback, off of every send socket.
         // Since we should never see data coming in on these sockets, having the receive complete means that
         // the socket is in an unknown state and we should close it and try again.
-        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
-        private static void ReceiveCallback(IAsyncResult result)
+        private static void ReceiveCallback(object sender, SocketAsyncEventArgs socketAsyncEventArgs)
         {
+            var t = socketAsyncEventArgs.UserToken as Tuple<Socket, IPEndPoint, SocketManager>;
```

Comment:
you mean in case `socketAsyncEventArgs` itself is null? That should not be possible... I would rather prefer that it throws in that case, as it would be completely unexpected